File: ListView-Demo/app/src/main/java/com/example/listviewdemo/MainActivity.java
package com.example.listviewdemo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ListView myListView = (ListView) findViewById(R.id.listView);
        final ArrayList<String> myGroup = new ArrayList<String>();
        myGroup.add("Vishal");
        myGroup.add("Connor");
        myGroup.add("Babul");
        myGroup.add("Vijay");
        myGroup.add("Ritesh");
        myGroup.add("Faisalwa");
        myGroup.add("Chubey Neta");
        myGroup.add("Rahul");
        myGroup.add("Arpit");
        myGroup.add("Debjit");
        myGroup.add("Shivnath");
        myGroup.add("Bagchi");
        ArrayAdapter<String> arrayAdapter = new ArrayAdapter<String>(this,android.R.layout.simple_expandable_list_item_1,myGroup);
        myListView.setAdapter(arrayAdapter);
        myListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Log.i("name",myGroup.get(position));
                Toast.makeText(MainActivity.this, myGroup.get(position), Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: BroadcastReceivers/app/src/main/java/com/example/broadcastreceivers/MainActivity.kt
package com.example.broadcastreceivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val psr = PowerStateReceiver()
        val iFilter = IntentFilter().apply {
            addAction(Intent.ACTION_POWER_DISCONNECTED)
            addAction(Intent.ACTION_POWER_CONNECTED)
        }
        registerReceiver(psr,iFilter)
    }
    // Dynamic Broadcast Receiver (Implementation)
    inner class PowerStateReceiver: BroadcastReceiver() {
        override fun onReceive(context: Context?, intent: Intent?) {
            if(intent?.action == Intent.ACTION_POWER_CONNECTED){
                Toast.makeText(context,"Charger Connected",Toast.LENGTH_SHORT).show()
            }
            if(intent?.action == Intent.ACTION_POWER_DISCONNECTED){
                Toast.makeText(context,"Charger Disconnected",Toast.LENGTH_SHORT).show()
            }
            if(intent?.action == null) return
        }

    }
}